### PR TITLE
Add some documentation about plugins in collections

### LINKED ISSUE #62231: Unclear how to use inventory plugins in collections
##### SUMMARY

I've been unable to get inventory plugins working inside of collections. The inventory plugin in ansible/ansible always takes precedence.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

collections

##### ANSIBLE VERSION

```paste below
devel
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

I've got a inventory.gcp.yml file that I run with `ansible-inventory --list -i inventory.gcp.yml`. I've got a collection called google.cloud installed into my collections path. Here are the various things I've tried.

```yaml
plugin: gcp_compute
...
```
Result: Runs the ansible/ansible inventory plugin.

```yaml
collections:
- google.cloud
plugin: gcp_compute
...
```
Result: Runs the ansible/ansible inventory plugin

```yaml
plugin: google.cloud.gcp_compute
...
```

Result: Plugin not found, errors out.

(I've also tried with `google.cloud.plugins.inventory.gcp_compute` and `google.cloud.inventory.gcp_compute` to no avail)

```yaml

```

##### ACTUAL RESULTS

```paste below

```

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Fixes #62231

Add some documentation about things that require FQCN, limitations on inventory caching, and code sharing.

Vars plugin documentation is in #61078. Vars plugins in collections are ignored prior to #61078.

##### ISSUE TYPE
- Docs Pull Request

### CODE REVIEW

File: docs/docsite/rst/plugins/cache.rst [hunk 1]
```diff
@@ -43,7 +50,7 @@ A custom cache plugin is enabled by dropping it into a ``cache_plugins`` directo
 Enabling Inventory Cache Plugins
 --------------------------------
 
-Inventory may be cached using a file-based cache plugin (like jsonfile). Check the specific inventory plugin to see if it supports caching.
```
Comment (bcoca) [inline docs/docsite/rst/plugins/cache.rst:46]: do we have a ticket to enable using cache plugins from collections?

Comment (s-hertel) [reply docs/docsite/rst/plugins/cache.rst:46]: Yes, there is this https://github.com/ansible/ansible/pull/56469.

File: docs/docsite/rst/dev_guide/developing_plugins.rst [hunk 2]
```diff
@@ -416,35 +416,7 @@ For example test plugins, see the source code for the `test plugins included wit
...
-The parameters are:
-
- * loader: Ansible's DataLoader. The DataLoader can read files, auto-load JSON/YAML and decrypt vaulted data, and cache read files.
- * path: this is 'directory data' for every inventory source and the current play's playbook directory, so they can search for data in reference to them. ``get_vars`` will be called at least once per available path.
- * entities: these are host or group names that are pertinent to the variables needed. The plugin will get called once for hosts and again for groups.
-
-This ``get vars`` method just needs to return a dictionary structure with the variables.
-
-Since Ansible version 2.4, vars plugins only execute as needed when preparing to execute a task. This avoids the costly 'always execute' behavior that occurred during inventory construction in older versions of Ansible.
+Vars plugins are unsupported by collections in Ansible version 2.9.
```
Comment (samccann) [inline docs/docsite/rst/dev_guide/developing_plugins.rst]: Can you clarify why vars is removed entirely? I see the other PR (#61078) adds this documentation back. The problem I see - this PR can be backported to 2.9, but the other PR cannot. That would leave us with vars docs in 2.9 not having anything but 'not supported in collections'.

File: docs/docsite/rst/plugins/inventory.rst [hunk 3]
```diff
@@ -65,6 +80,8 @@ The ``auto`` inventory plugin is enabled by default and works by using the ``plu
       |  |--ec2-98-765-432-10.compute-1.amazonaws.com
       |--@ungrouped:
 
+If you are using an inventory plugin in a playbook-adjacent collection and want to test your setup with ``ansible-inventory``, you will need to use the --playbook-dir flag.
```
Comment (samccann) [inline docs/docsite/rst/plugins/inventory.rst]:
```suggestion
If you are using an inventory plugin in a playbook-adjacent collection and want to test your setup with ``ansible-inventory``, you will need to use the ``--playbook-dir`` flag.
```

File: docs/docsite/rst/dev_guide/developing_plugins.rst (hunk 2 above)
Comment (s-hertel) [reply docs/docsite/rst/dev_guide/developing_plugins.rst]: Oh, oops! I thought this was in the docs/docsite/rst/dev_guide/developing_collections.rst file when I removed that. I'll put that back, thank you for catching it.

Comment (s-hertel) [reply docs/docsite/rst/dev_guide/developing_plugins.rst]: Fixed, sorry about that.

**Review by samccann — APPROVED** [review]

**Pull request merged**